pkg/controller/knativeserving/openshift/openshift.go

```diff
@@ -556,3 +573,76 @@ func configureLogURLTemplate(u *unstructured.Unstructured) error {
 	}
 	return nil
 }
+
+func exposeMetricToServiceMonitor(instance *servingv1alpha1.KnativeServing) error {
+	// namespace where knative-serving-operator has been installed
+	namespace, err := getNamespace()
+	if err != nil {
+		log.Info(fmt.Sprintf("empty %s because operator dont want to expose metric or running locally", knativeServingInstalledNamespace))
+		return nil
+	}
+
+	log.Info("exposing metric for installed knative version")
+	knativeVersion.WithLabelValues(version.Version, namespace, instance.GetName()).Set(float64(time.Now().Unix()))
+
+	// Get service knative-serving-operator in order to create service monitor
+	svcDetail := getSvcDetails(namespace)
+	if svcDetail == nil {
+		log.Info(fmt.Sprintf("service knative-serving-operator does not exist in %s namespace", namespace))
+		return nil
+	}
+
+	// Add label openshift.io/cluster-monitoring to namespace
+	if err = labelNamespace(namespace); err != nil {
+		return err
+	}
+
+	// CreateServiceMonitors creates ServiceMonitors objects based on Service objects.
+	if _, err := metric.CreateServiceMonitors(cfg, namespace, []*v1.Service{svcDetail}); err != nil {
+		if strings.Contains(err.Error(), "already exists") {
+			log.Info(fmt.Sprintf("ServiceMonitor already exist for %s service", svcDetail.Name))
+			return nil
+		}
+		log.Error(err, "Unable to create ServiceMonitor")
+		return err
+	}
+	log.Info("successfully created ServiceMonitor")
+	return nil
+}
+
+func getSvcDetails(ns string) *corev1.Service {
+	svc := &corev1.ServiceList{}
+	if err := api.List(context.TODO(), &client.ListOptions{Namespace: ns}, svc); err != nil {
+		return nil
+	}
+	for i := range svc.Items {
+		if svc.Items[i].Name == "knative-serving-operator" {
+			return &svc.Items[i]
+		}
+	}
+	return nil
+}
+
+func getNamespace() (string, error) {
+	ns, found := os.LookupEnv(knativeServingInstalledNamespace)
+	if !found {
+		return "", fmt.Errorf("%s must be set", knativeServingInstalledNamespace)
+	}
+	return ns, nil
+}
+
+func labelNamespace(namespace string) error {
```

Comment:
Can we name this `addMonitoringLabelToNamespace` or something? `labelNamespace` sounds more generic than it is.